app/src/components/geometries.tsx

```diff
@@ -491,11 +507,147 @@ function GeometryComponent({
+          position={geometry.position}
+          rotation={geometry.rotation}
+          scale={[
+            geometry.scale[0] * geometry.a,
+            geometry.scale[1] * geometry.b,
+            geometry.scale[2] * geometry.c,
+          ]}
+          onPointerMove={onPointerMove}
+          onPointerOver={onPointerOver}
+          onPointerOut={onPointerOut}
+        >
+          <meshStandardMaterial
+            attach="material"
+            color={geometry.material.color}
+            opacity={geometry.material.opacity}
+            wireframe={geometry.material.wireframe}
+            transparent={geometry.material.opacity < 1.0}
+          />
+          {geometry.material.outlines && (
+            <Outlines
+              thickness={0.05}
+              color={geometry.material.color}
+              opacity={geometry.material.opacity}
+            />
+          )}
+        </Sphere>
+      );
+    case "Rhomboid": {
+      const [rhomboidGeometry, setRhomboidGeometry] =
+        useState<THREE.BufferGeometry | null>(null);
+
+      useEffect(() => {
+        const vectorA = geometry.vectorA;
+        const vectorB = geometry.vectorB;
+        const vectorC = geometry.vectorC;
+
+        const vertices = [
+          new THREE.Vector3(0, 0, 0),
+          new THREE.Vector3(...vectorA),
+          new THREE.Vector3(...vectorB),
+          new THREE.Vector3(...vectorA).add(new THREE.Vector3(...vectorB)),
+          new THREE.Vector3(...vectorC),
+          new THREE.Vector3(...vectorA).add(new THREE.Vector3(...vectorC)),
+          new THREE.Vector3(...vectorB).add(new THREE.Vector3(...vectorC)),
+          new THREE.Vector3(...vectorA)
+            .add(new THREE.Vector3(...vectorB))
+            .add(new THREE.Vector3(...vectorC)),
+        ];
+
+        const positions = new Float32Array(
+          vertices.flatMap((v) => [v.x, v.y, v.z]),
+        );
+
+        const indices = [
+          0,
+          2,
+          1,
+          1,
+          2,
+          3, // Bottom face
+          4,
+          5,
+          6,
+          5,
+          7,
+          6, // Top face
+          0,
+          1,
+          4,
+          1,
+          5,
+          4, // Front face
+          1,
+          3,
+          5,
+          3,
+          7,
+          5, // Right face
+          3,
+          2,
+          7,
+          2,
+          6,
+          7, // Back face
+          2,
+          0,
+          6,
+          0,
+          4,
+          6, // Left face
+        ];
+
+        const newRhomboidGeometry = new THREE.BufferGeometry();
+        newRhomboidGeometry.setAttribute(
+          "position",
+          new THREE.BufferAttribute(positions, 3),
+        );
+        newRhomboidGeometry.setIndex(indices);
+        newRhomboidGeometry.computeVertexNormals();
+
+        setRhomboidGeometry(newRhomboidGeometry);
+      }, [geometry]);
+
+      if (!rhomboidGeometry) return null;
```

Comment:
Can you include this in a single return statement? Something like 
{rhomboidGeometry && ( <>...</>)} instead of this extra return

Reply:
The code I updated with the recent commit should return `null` if there is no geometry. Is this alright with you?

Reply:
That was the default before as well, right? Should be good.